Gopkg.toml

```diff
@@ -0,0 +1,118 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+
```

Comment:
I am not sure this file is equivalent to glide.yaml. We can debate separately if we do or do not want to pin certain things, but for the purpose of this migration I would prefer no changes to the dependencies.

Specifically, `dep` defaults to `^`, i.e. `version = "1.16.0"` in `dep` is equivalent to `"^1.16.0"` in glide. In order to do an actual pin, we need to use `=`, e.g. `version = "=1.16.0"`

Reply:
I don't think it did, none of the projects are pinned, while many were pinned in glide.

Reply:
OK but why did we pin these. I imagine this will end up being a nuisance for internal dep to handle constraints.

Reply:
we don't pin all of them, but we do pin some, esp. I remember having issues with gogo versions and had to work out a consistent working set. Like I said, we should have a discussion about not-pinning, but this migration should not be introducing changes to dependencies, imo.

Reply:
It is funny you say that because internally gogo/protobuf seems to be the packages causing the issue I have experienced.